### PR TITLE
use a markdown linter on new files

### PR DESCRIPTION
**Pull request by mahf708**
Adds a markdown linter on new files in PRs to ensure they conform with some linting rules (can modify these later if needed).

[BFB]

### CODE REVIEW

File: .github/workflows/e3sm-gh-ci-cime-tests.yml [hunk 1]
```diff
@@ -3,6 +3,11 @@ name: gh
 on:
   pull_request:
     branches: [ master ]
+    paths:
+      - 'cime_config/**'
+      - 'components/eam/**'
+    paths-ignore:
+      - 'components/eam/docs/**'
```
Comment (rljacob) [inline .github/workflows/e3sm-gh-ci-cime-tests.yml:10]: F-cases also test the land model so should include components/elm/**

**Comment by rljacob** [comment]
The linter will only run on new or modified files?

**Review by bartgol — APPROVED** [review]
Where will the linter report? Do we just open the action log and see some output from it?

**Comment by rljacob** [comment]
Good I want it to just look at new/modified files.  Don't fix any old ones.

**Comment by rljacob** [comment]
How do I run this in my docs development environment on my laptop?

**Comment by rljacob** [comment]
pre-commit hook would be good.  Can you add one to https://github.com/E3SM-Project/E3SM-Hooks  ?

**Comment by rljacob** [comment]
Actually the hooks in that repo are bash-based hooks that run anywhere.  It looks like the pre-commit hook for the linter needs python and some things installed.

File: .github/workflows/e3sm-gh-ci-cime-tests.yml [hunk 2]
```diff
@@ -3,6 +3,16 @@ name: gh
 on:
   pull_request:
     branches: [ master ]
+    paths:
+      # first, yes to these
+      - 'cime_config/**'
+      - 'components/eam/**'
+      - 'components/elm/**'
```
[the remark this answers is not in the document]
Comment (rljacob) [reply .github/workflows/e3sm-gh-ci-cime-tests.yml:10]: Yes do that and also include edits to the coupler.

**Comment by bartgol** [comment]
This action already blocks the PR merging, right? If so, I'm ok if with an action (rather than a hook), forcing the dev to do a bit of work.

Quick question for @mahf708 : does this action only run on _new_ files? What if someone modifies an existing one, using bad formatting?

File: .github/workflows/e3sm-gh-ci-cime-tests.yml [hunk 3]
```diff
@@ -3,16 +3,6 @@ name: gh
 on:
   pull_request:
     branches: [ master ]
-    paths:
```
Comment (bartgol) [inline .github/workflows/e3sm-gh-ci-cime-tests.yml]: Why did you remove these? If a PR only touches documentation, it seems pointless to run the CIME tests..

**Review by bartgol — APPROVED** [review]

File: .github/workflows/e3sm-gh-ci-cime-tests.yml (hunk 3 above)
Comment (rljacob) [reply .github/workflows/e3sm-gh-ci-cime-tests.yml]: Those went in a different PR.

Comment (bartgol) [reply .github/workflows/e3sm-gh-ci-cime-tests.yml]: Ah, weird. GH showed them to me as part of the changes since my last review. But indeed it doesn't show them when I look for all changes.

Thanks!
